docs/plugins.md

```diff
@@ -1,5 +1,202 @@
+
+$retryPlugin = new RetryPlugin();
+$redirectPlugin = new RedirectPlugin();
+```
+
+Then you can create a `PluginClient`:
+
+```php
+use Http\Discovery\HttpClientDiscovery;
+use Http\Client\Plugin\PluginClient;
+
+...
+
+$pluginClient = new PluginClient(HttpClientDiscovery::find(), [
+    $retryPlugin,
+    $redirectPlugin
+]);
+```
+
+After you can use this plugin client like a classic `Http\Client\HttpClient` or `Http\Client\HttpAsyncClient` one:
+
+```php
+// Send a request
+$response = $pluginClient->sendRequest($request);
+
+// Send an asynchronous request
+$promise  = $pluginClient->sendAsyncRequest($request);
+```
+
+Read the [tutorial](tutorial.md) to know more about `HttpClient` and `HttpAsyncClient`
+
+## Available plugins
+
+Each plugin have its own configuration and dependencies, check the documentation for each of the available plugin:
+
+ - [Authentication](plugins/authentication.md): Add authentication header on a request
+ - [Cookie](plugins/cookie.md): Add cookie to request and save them from the response by using a CookieJar
+ - [Encoding](plugins/encoding.md): Add support for receiving chunked, deflate or gzip response 
+ - [Error](plugins/error.md): Transform bad response (400 to 599) to exception
+ - [Redirect](plugins/redirect.md): Follow redirection coming from 3XX responses
+ - [Retry](plugins/retry.md): Retry a failed call
+ - [Stopwatch](plugins/stopwatch.md): Log time of a request call by using the Stopwatch component
+
+## Order of plugins
+
+When you inject an array of plugins into the `PluginClient`, order of the plugins in array matters.
+
+Plugins are transformed into a plugin chain, where the first element in array will be called first, and, 
+obviously, the last element will be called at the end. However when the `ResponseInterface` or the `Promise` is
+received from the underlying client, this execution chain is revert and the last plugin defined will be called first.
+
+i.e. with the following code: 
+
+```php
+use Http\Discovery\HttpClientDiscovery;
+use Http\Client\Plugin\PluginClient;
+use Http\Client\Plugin\RetryPlugin;
+use Http\Client\Plugin\RedirectPlugin;
+
+$retryPlugin = new RetryPlugin();
+$redirectPlugin = new RedirectPlugin();
+
+$pluginClient = new PluginClient(HttpClientDiscovery::find(), [
+    $retryPlugin,
+    $redirectPlugin
+]);
+```
+
+The execution chain will look like this:
+
+```
+Request  ---> PluginClient ---> RetryPlugin ---> RedirectPlugin ---> HttpClient ----
+                                                                                   | (processing call)
+Response <--- PluginClient <--- RetryPlugin <--- RedirectPlugin <--- HttpClient <---
+```
+
+In order to have correct behavior over the global process, you need to understand well each plugin used,
+and manage a correct order when passing the array to the `PluginClient`
+
+`RetryPlugin` will be best at then for example if you want to optimize the retry process, but also it can be good 
```

Comment:
Something missing from this sentence?

Also in the first? Then it can be the last as well?